### DISCUSSION TITLE
tutorials/lift-off-part1/08-apollo-client-setup

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#33] [course feedback]
# tutorials/lift-off-part1/08-apollo-client-setup

https://www.apollographql.com/tutorials/lift-off-part1/08-apollo-client-setup

**Comment by VihoSerge** [comment]
I didn't see the use of the `graphql` package. was it necessary to install it ?

**Reply by mabuyo** [reply]
Great question @VihoSerge! Yes, we do need to install the `graphql` package for Apollo Client to work. Although we don't import the package and use it _directly_ anywhere in our own files, Apollo Client is using it under the hood!

**Reply by VihoSerge** [reply]
Thanks @mabuyo. it's clear now

**Reply by anvodev** [reply]
Thanks @mabuyo, also thanks for the question @VihoSerge

**Comment by davidfeldt** [comment]
Love this tutorial - it's very well done

**Reply by le-el** [reply]
I agree

**Comment by esweinman** [comment]
I think your code challenge is a little confusing, it says to set up a new instance to that points to a specific endpoint but does not ask to include stored results via caching. However the solution requires caching to be correct. I would rewrite the challenge to be more clear. I spent more time than necessary on this challenge because of it. Thanks!

**Reply by mabuyo** [reply]
Thanks for the feedback @esweinman ! We'll update this code challenge shortly to be more clear about what we're asking for.

**Comment by Ahmedhossamdev** [comment]
there is a problem with editor `loading editor`

**Reply by mabuyo** [reply]
Hey there @Ahmedhossamdev - sorry for the delay. Where are you seeing this error pop up?

**Comment by AnithaSureshkumar** [comment]
editor is keep loading....

**Reply by mabuyo** [reply]
Hey there @AnithaSureshkumar - can you please clarify where you're seeing this error pop up?

**Reply by AnithaSureshkumar** [reply]
loading editor (its keep loading), other then that no msg.. i left this challenge and i m continuing.. but wherever editor there in challenge now i m facing same issue.. could you help me

**Reply by mabuyo** [reply]
Thanks for the info - are you talking about the code challenge in the lesson (see screenshot below)? Would you be able to provide a screenshot of what you're seeing, or any messages in the developer console?

<img width="1005" alt="Screenshot 2023-09-01 at 9 56 27 AM" src="https://github.com/apollographql-education/content-feedback/assets/6900407/40ae0a1e-876b-4b71-b6be-2f544d21014f">

**Reply by AnithaSureshkumar** [reply]
yes same ..

**Reply by mabuyo** [reply]
Hey @AnithaSureshkumar - would you be able to provide a screenshot of what you're seeing, or any messages in the developer console?

**Comment by AnsBdran** [comment]
why do we need the `graphql` package on the front-end, doesn't the server handles parsing `graphql` queries and respond to these queries?

**Reply by mabuyo** [reply]
Hey @AnsBdran! Yes, we do need to install the `graphql` package for Apollo Client to work. Although we don't import the package and use it directly anywhere in our own files, Apollo Client is using it under the hood!

**Comment by itsramiel** [comment]
In the code challenge, it would be clearer if it was mentioned to assign the result to a consant name `client` making the text:
```
Create a new ApolloClient instance assigning the result to a constant name `client`, with its options set up to connect to the endpoint https://graphql.org/swapi-graphql and use the InMemoryCache.

**Reply by mabuyo** [reply]
Good point @itsramiel - thanks for bringing this up! We'll make those changes for clarity.

**Comment by marksikaundi** [comment]
Over time am getting hand of it very well

**Comment by Peter-AMD** [comment]
There is an ERESOLVE error when installing graphql and @apollo/client in the frontend.
Updating react to v18 solves this issue.
`npm install react@latest react-dom@latest`

**Comment by belloshehu** [comment]
So lovely! The interactive code challenge makes learning fun and helps to solidify understanding! Great experience!

**Comment by repiemm** [comment]
I have struggled only with the concepts of React's context API, and React hooks, but I've researched about them and I think I've understood enough to proceed to the next step.

**Comment by NishantCoder108** [comment]
# Apollo Client Setup 💪

- install @apollo/client and graphql
- ApolloClient ,ApolloProvider , and InMemoryCache , that will come from "@apollo/client"
- we make new instance of ApolloClient
   ```
    const client = new ApolloClient({
     uri :"https://example.com",
     cache: new InMemoryCache()
    })

- Finally client will pass as a props in Apollo Provider


### Note : Apollo/Client ---> it is used for caching , local state management , error handling  and 
Graphql is used for parsing the graphql queries

**Comment by sharmun** [comment]
npm install graphql @apollo/client -D
react and @apollo/client should be in the same node_modules folder

**Comment by felabel** [comment]
This is one of the best tutorials, started from the docs and now i am enjoying it. Thank you so much team!

**Comment by alison-hernandez** [comment]
To my fellow Apollo Client 4 friends, I was able to complete this tutorial successfully with a different set of imports and the updated format for ApolloClient:

```
import {
  ApolloClient,
  HttpLink,
  InMemoryCache
} from '@apollo/client';
import {ApolloProvider} from "@apollo/client/react";

const client = new ApolloClient({
  link: new HttpLink({ uri: 'http://localhost:4000' }),
  cache: new InMemoryCache()
})
```
